src/interface/mesh_data.hpp

```diff
@@ -37,21 +38,67 @@ template <typename T>
 using BlockDataList_t = std::vector<std::shared_ptr<MeshBlockData<T>>>;
 
 namespace pack_on_mesh_impl {
+
+template <typename K>
+inline void AppendKey(K *key_collection, const K *new_key);
+
+template <>
+inline void AppendKey<std::vector<std::string>>(std::vector<std::string> *key_collection,
+                                                const std::vector<std::string> *new_key) {
+  for (const auto &k : *new_key) {
+    key_collection->push_back(k);
+  }
+}
+
+template <>
+inline void AppendKey<vpack_types::StringPair>(vpack_types::StringPair *key_collection,
+                                               const vpack_types::StringPair *new_key) {
+  for (const auto &k : new_key->first) {
+    key_collection->first.push_back(k);
+  }
+  for (const auto &k : new_key->second) {
+    key_collection->second.push_back(k);
+  }
+}
+
```

Comment:
Yes, both function are called with single key that is added to the collection of keys. However, in the first case a "key" is a string, in the second case, it's a pair of two strings. The template specialization is necessary here so that we can reuse the code for making a variable pack (where keys are strings) and combined Flux-Variable packs (where keys are pairs of strings). I'll add a clarifying comment to these specializations.

Reply:
Oh, I see the confusion now. I guess you're right, `new_key` is multiple keys, namely all the individual keys of a (flux-)variable pack, which together make up the singular key of that pack. So it's more like a meta-key...

But yeah, I'll rename the function and arguments.

Reply:
Actually, maybe it's more consistent to stick with the singular version, because in the `PackVariables` interface we also call the entire vector of strings `key` and not `keys`. And indeed, the underlying `std::map`s use `std::vector<std::string>` pair of vector strings as keys.

Reply:
So technically it's an `AppendKeyVector` function ;) -- just kidding, sounds good. Let's then keep it as is.